sutta_publisher/src/sutta_publisher/edition_parsers/epub.py

```diff
@@ -70,26 +70,37 @@ def __make_chapter_index(
             return _links
 
     def __make_chapter(
-        self, html: BeautifulSoup, chapter_number: int, section_name: str = ""
-    ) -> Tuple[epub.EpubHtml, List[epub.Link]]:
+        self, html: BeautifulSoup, chapter_number: int, section_name: str = "", make_index: bool = True
+    ) -> Tuple[epub.EpubHtml, List[epub.Link] | None]:
         file_name = f"chapter_{chapter_number}.xhtml"
         chapter = self.__make_chapter_content(html, file_name)
-        index = _chapter = self.__make_chapter_index(html, file_name, section_name=section_name)
+
+        index = _chapter = self.__make_chapter_index(html, file_name, section_name=section_name) if make_index else None
+
         return chapter, index
 
     def __set_chapters(
-        self, book: epub.EpubBook, html: BeautifulSoup, chapter_number: int, section_name: str = ""
+        self, book: epub.EpubBook, html: BeautifulSoup, chapter_number: int, section_name: str = "", make_index=True
     ) -> None:
-        chapter, index = self.__make_chapter(html=html, chapter_number=chapter_number, section_name=section_name)
+        chapter, index = self.__make_chapter(
+            html=html, chapter_number=chapter_number, section_name=section_name, make_index=make_index
+        )
         book.add_item(chapter)
-        book.toc.extend(index)
+        if index is not None:
+            book.toc.extend(index)
         book.spine.append(chapter)
 
     def __generate_epub(self) -> None:
         """Generate epub"""
         log.debug("Generating epub...")
 
         _volumes_in_html = [BeautifulSoup(_, "lxml") for _ in self._EditionParser__generate_html()]  # type: ignore
+        frontmatters = [BeautifulSoup(_, "lxml") for _ in self._EditionParser__generate_frontmatter().values()]
+
+        volume_number = 0
+
+        # for _html in _volumes_in_frontmatter:
```

Comment:
there is this iterating and incrementing in lines 116-118 so probably you should remove the commented out code